File: src/main/resources/archetype-resources/src/test/java/__className__Test.java
package ${package};

import org.bonitasoft.engine.connector.ConnectorException;
import org.bonitasoft.engine.connector.ConnectorValidationException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class ${className}Test {

    ${className} connector;

    @BeforeEach
    void setUp() {
        connector = new ${className}();
    }

    @Test
    void should_throw_exception_if_mandatory_input_is_missing() {
        assertThrows(ConnectorValidationException.class, () ->
                connector.validateInputParameters()
        );
    }

    @Test
    void should_throw_exception_if_mandatory_input_is_empty() {
        Map<String, Object> parameters = new HashMap<>();
        parameters.put(${className}.DEFAULT_INPUT, "");
        connector.setInputParameters(parameters);
        assertThrows(ConnectorValidationException.class, () ->
                connector.validateInputParameters()
        );
    }

    @Test
    void should_throw_exception_if_mandatory_input_is_not_a_string() {
        Map<String, Object> parameters = new HashMap<>();
        parameters.put(${className}.DEFAULT_INPUT, 38);
        connector.setInputParameters(parameters);
        assertThrows(ConnectorValidationException.class, () ->
                connector.validateInputParameters()
        );
    }

    @Test
    void should_create_output_for_valid_input() throws ConnectorException {
        Map<String, Object> parameters = new HashMap<>();
        parameters.put(${className}.DEFAULT_INPUT, "valid");
        connector.setInputParameters(parameters);
        Map<String, Object> outputs = connector.execute();
        assertThat(outputs.get("defaultOutput")).isEqualTo("valid - output");
    }

}

File: src/main/resources/archetype-resources/src/test/kotlin/__className__Test.kt
package ${package}

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.bonitasoft.engine.connector.ConnectorValidationException
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class ${className}Test {

    lateinit var connector: ${className}

    @BeforeEach
    fun setUp() {
        connector = ${className}();
    }

    @Test
    fun `should throw exception if mandatory input is missing`() {
        assertThatThrownBy { connector.validateInputParameters() }
                .isExactlyInstanceOf(ConnectorValidationException::class.java)
    }

    @Test
    fun `should throw exception if mandatory input is empty`() {
        // Given
        connector.setInputParameter(${className}.DEFAULT_INPUT, "")

        // When
        assertThatThrownBy { connector.validateInputParameters() }
                // Then
                .isExactlyInstanceOf(ConnectorValidationException::class.java)
    }

    @Test
    fun `should throw exception if mandatory input is not a string`() {
        // Given
        connector.setInputParameter(${className}.DEFAULT_INPUT, 38);

        // When
        assertThatThrownBy { connector.validateInputParameters() }
                // Then
                .isExactlyInstanceOf(ConnectorValidationException::class.java)
    }

    @Test
    fun `should create output for valid input`() {
        // Given
        connector.setInputParameter(${className}.DEFAULT_INPUT, "valid");

        // When
        connector.executeBusinessLogic();

        // Then
        val output = connector.getOutputParameter(${className}.DEFAULT_OUTPUT);
        assertThat(output).isEqualTo("valid - output");
    }
}